### PR TITLE
feat: compact design

### PR DESCRIPTION
**Pull request by MAbdurrehman12**
#### Pre-Flight checklist

- [ ] Screenshots and design review for any changes that affect layout or styling
  - [ ] Desktop screenshots
  - [ ] Mobile width screenshots
- [ ] Testing
  - [ ] Code is tested
  - [ ] Changes have been manually tested

#### What are the relevant tickets?
Closes https://github.com/mitodl/ocw-hugo-themes/issues/765

#### What's this PR do?
- Creates a new theme `course-v2` and implements new designs as provided in the issue ticket.

#### How should this be manually tested?
- Checkout this branch
- In your local ocw course config (having env: `COURSE_HUGO_CONFIG_PATH`), add/rename `course` to `course-v2` to use this new theme instead of the previous one.
- Build any course locally.
- Verify that the UI is according to the provided designs for the course home page and course interior pages.
- Repeat this for:
    - as many courses as you can :) I'm also currently testing.
    - multiple browsers (chrome, firefox, safari etc)
    - multiple screen sizes (desktop, laptop, tablet, mobile etc)
- **Important:** Verify that the previous/old design of `course` and `www` still works as expected and there are no side-effects/impacts. 

#### Screenshots
- Please see this [deployed course](https://ocw-hugo-themes-course-pr-784--ocw-next.netlify.app/).

### CODE REVIEW

**Comment by pdpinch** [comment]
@Wassaf-Shahzad how/can I see this PR deployed with a course?

**Comment by MAbdurrehman12** [comment]
> how/can I see this PR deployed with a course?

@pdpinch This is a [deployed course](https://ocw-hugo-themes-course-pr-784--ocw-next.netlify.app/). But the PR is not fully ready yet so there are things that will not seem right yet.

**Comment by Wassaf-Shahzad** [comment]
> 🚀 Deployed on https://ocw-hugo-themes-course-pr-784--ocw-next.netlify.app

@pdpinch This one

**Comment by MAbdurrehman12** [comment]
@pdpinch Currently, I'm testing multiple courses locally and trying to find issues but I think you can also take a look and if anyone wants to review the code, that is fine too.

File: course-v2/layouts/partials/course_info.html [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+      </div>
+      {{ end }}
+    </div>
+  </div>
+</div>
+
+<script>
+// todo: move this function to js file.
+// currently having some issues while using this function from js file
+function desktopCourseInfoToggle(){
```
Comment (MAbdurrehman12) [inline course-v2/layouts/partials/course_info.html:95]: I'm making a follow-up PR for https://github.com/mitodl/ocw-hugo-themes/issues/795 and will change this approach along with moving this code to a suitable place.

**Comment by Wassaf-Shahzad** [comment]
@pdpinch Should there be better spacing between table headings and rows

Screenshot 2022-07-19 at 6 00 20 PM

**Comment by pdpinch** [comment]
> @pdpinch Should there be better spacing between table headings and rows

I don't think the designs include any updates to the table headings and rows. 

@Wassaf-Shahzad what are you suggesting we should change?

**Comment by Wassaf-Shahzad** [comment]
> > @pdpinch Should there be better spacing between table headings and rows
> 
> I don't think the designs include any updates to the table headings and rows.
> 
> @Wassaf-Shahzad what are you suggesting we should change?



> #765

what I am inquiring is whether we should increase the spacing between table row and table heading as they currently looked a bit squished (as shown ) on the screenshot above.

**Review by Wassaf-Shahzad — CHANGES_REQUESTED** [review]
Looks and works great ! Just some minor things
- Some files dont end with a new line, those should be added
- Tables are not responsive, I checked this behaviour upto 320px which is the recommended smallest screen size. Screen shots attached
- Footer also behaves differently on small screens

### Tables on ocwnext
Screenshot 2022-07-20 at 3 34 59 PM

### Tables on course-v2
Screenshot 2022-07-20 at 3 47 03 PM

**Comment by gumaerc** [comment]
@Wassaf-Shahzad @MAbdurrehman12  In the console you see this error:

```
(index):1814 Uncaught TypeError: Cannot read properties of null (reading 'getElementsByTagName')
    at initResponsiveTables ((index):1814:80)
    at (index):1814:512
```

It seems this is because the script that does this was looking for the div with the `id` of `main-content`, which doesn't seem to exist anymore.  Instead it seems that a div with the class `course-home-content` is the top level div in the content area, even when you're not on the course home page.  We need to assign an ID again to the outer element in the content area and make sure it matches what's in the script.

image

**Comment by MAbdurrehman12** [comment]
Thanks for looking into this @gumaerc @Wassaf-Shahzad 
I'll update the PR tomorrow

**Comment by Wassaf-Shahzad** [comment]
LGTM ! but you should still wait for @gumaerc if he has any input

**Review by gumaerc — CHANGES_REQUESTED** [review]
Overall this looks very good.  As I noted in https://github.com/mitodl/ocw-hugo-themes/pull/803, it seems that we have not isolated the changes to the header and footer to `course-v2.`  I think we'll want to create v2 versions of both the header and the footer in `base-theme` and use those in `course-v2` instead.  You can see that when I use this branch locally while only specifying `["course", "base-theme"]` as  my themes in my config, I get the new header and footer when using the old theme:

image

image

vs live production:

image

image

**Comment by MAbdurrehman12** [comment]
@gumaerc instead of creating separate files for header and footer, I have overridden some CSS in course-v2 theme as I had some issues in creating separate files and this approach (of overriding CSS) also looked quite simple.
I have also added this comment that when we completely roll out to new theme then we can get rid of this overridden CSS.

Apart from this, the new theme also has a different logo, which I did not change/replace with the old one as this logo is much clearer.

Can you please have another look? and let me know if you find out any issue.

Thanks

**Review by gumaerc — CHANGES_REQUESTED** [review]
The CSS changes are a fine approach.  There are only a couple other things I noticed:

 - The logo image has been replaced with a smaller version and the original version has not been preserved, so you get the smaller logo image when rendering with the old theme:
 
image

vs

image

 - You've been doing a good job commenting, but it would be nice if we could leave a marker, something like `TODO: course-v2` wherever inline code changes like the header / footer CSS will need to be reconciled when it comes time to merge this into `course`.  That way, we have a consistent tag we can search for to easily locate all the places where we couldn't simply just override the existing layout in the `course-v2` folder.

**Review by gumaerc — APPROVED** [review]
:+1: LGTM!

**Comment by pdpinch** [comment]
Do we not have tests for the responsive function?

**Pull request merged**